phpBB/includes/functions_content.php

```diff
@@ -405,12 +405,38 @@ function decode_message(&$message, $bbcode_uid = '')
 		$replace = array("\n");
 	}
 
+	/**
+	* Use this event to modify the message before it is decoded
+	*
+	* @event core.decode_message_before
+	* @var string	message			The message content
+	* @var string	bbcode_uid		The message BBCode UID
+	* @var string	match			Match pattern to replace
+	* @var int		replace			Replacement for the matched text
+	* @since 3.1.8-RC1
+	*/
+	$vars = array('message', 'bbcode_uid', 'match', 'replace');
```

Comment:
```$message``` is passed by reference. Please use an alternative variable for the event. Also, the match and replace will no longer properly work for 3.2.